tests/test_lowlevel.py

```diff
@@ -233,11 +232,6 @@ def redirect_resp_handler(sock):
         assert r.status_code == 200
         assert len(r.history) == 1
         assert r.history[0].status_code == 301
-
-        # currently Python3 not handling non-ASCII redirects (issue #3888)
-        if is_py3:
-            assert redirect_request[0].startswith(b'GET /' + expected_path_py3 + b' HTTP/1.1')
-        else:
-            assert redirect_request[0].startswith(b'GET /' + expected_path + b' HTTP/1.1')
+        assert redirect_request[0].startswith(b'GET /' + expected_path + b' HTTP/1.1')
```

Comment:
We're kind of testing the test here. Perhaps also adding a check that `r.history[0].url` is what we expect would be helpful?